### PR TITLE
Pulls/1/fix delete uploaded files

### PR DESCRIPTION
**Pull request by phalkunz**
Issue: https://github.com/silverstripe/silverstripe-asset-admin/issues/699

---

Refer to @unclecheese's comment for more details https://github.com/silverstripe/silverstripe-asset-admin/issues/699#issuecomment-352634380

### CODE REVIEW

File: client/src/containers/Gallery/Gallery.js [hunk 1]
```diff
@@ -102,14 +102,6 @@ class Gallery extends Component {
 
       $select.off('change');
     }
-
-    // If props.files has changed, flush any uploaded files.
-    // The render() logic will merge queuedFiles into the props.files array
-    // until this is called, leaving completed uploads temporarily in the current view,
-    // even if they're technically not part of props.files.
-    if (!this.compareFiles(this.props.files, nextProps.files)) {
```
Comment (flamerohr) [inline client/src/containers/Gallery/Gallery.js]: I think this is the only use of `compareFiles()` which means it's no longer needed.

Could we remove it and associated tests?

Comment (unclecheese) [reply client/src/containers/Gallery/Gallery.js]: Yeah, good catch. That function is showing its age.

Comment (tractorcow) [reply client/src/containers/Gallery/Gallery.js]: When is purgeUploadQueue() called now?

Comment (tractorcow) [reply client/src/containers/Gallery/Gallery.js]: What about if folder ID changes? It looks like it only happens on sort otherwise.

Comment (phalkunz) [reply client/src/containers/Gallery/Gallery.js]: @tractorcow `purgeUploadQueue()` is not being called from anywhere at the moment but @unclecheese said to keep it. I can't answer your second question but I'm sure @unclecheese can.

**Comment by tractorcow** [comment]
As discussed at https://github.com/silverstripe/silverstripe-asset-admin/issues/699#event-1396843654 the only needed fix was the ID cast. Is that sufficient?

**Comment by flamerohr** [comment]
Yup, lets do that, can we fix the casting on the server-side?

client-side is fine, but this leaves a tiny bit of tech debt for future development

**Comment by tractorcow** [comment]
I've switched to 4.0 branch and switched the files comparison for a simple folder id check. :)

**Comment by flamerohr** [comment]
@tractorcow nice one, but that still doesn't resolve my original comment :)
`compareFiles` is no longer needed/used, so it could be removed.

**Comment by tractorcow** [comment]
Yeah but removing methods is semver breakage, so I've deprecated it instead.

**Comment by tractorcow** [comment]
Ok I've removed it after confirming Gallery.js isn't public API. :)

**Pull request merged**